src/search/Search.tsx

```diff
@@ -99,30 +97,30 @@ export function Search({ controller, onClose }: SearchProps) {
             onClick={(event) => {
                 const target = event.target as HTMLElement;
                 if (target.id == "search-dialog") { 
-                    _onClose() 
+                    onCloseRequest()
                 }
                 const LI = target.closest("li");
                 if (LI != null) {
                     const index = LI.getAttribute("data-index");
                     if (index) {
                         results[parseInt(index)].action();
-                        setTimeout(_onClose, 1);
+                        setTimeout(onCloseRequest, 1);
                     }
                 }
             }}
-            className={`fixed z-50 inset-0 bg-gray-900 bg-opacity-60 overflow-y-auto h-full w-full px-4 ${visible ? '' : 'hidden'}`}>
+            className={`fixed z-50 inset-0 bg-gray-900 bg-opacity-60 overflow-y-auto h-full w-full px-4`}>
             <div className="relative top-20 mx-auto shadow-xl rounded-md bg-zinc-700 max-w-md p-1 flex flex-col">
-                {controller.prompt.length > 0 ?
+                {prompt.length > 0 ?
```

Comment:
IMO prompt to powinien być String | null, sprawdzanie długości stringa jest troche krzywe :D

```
{prompt &&  <h1 className="p-1 text-lg py-2">{prompt}</h1>}
                    ```

Reply:
To najpierw trzeba wiedzieć jak to napisać :D 
Poprawiłem